File: frontend/role-rolls/src/app/campaigns/models/RollInput.ts
import { Property } from '@app/models/bonuses/bonus';
import {Creature} from '@app/campaigns/models/creature';

export interface RollInput {
  propertyName: string;
  propertyValue: number;
  property: Property;
  attribute?: Property | null;
  numberOfDices: number
  complexity: number;
  difficulty: number;
  advantage: number;
  luck: number;
  bonus: number;
  hidden: boolean;
  creature: Creature;
  rollsAsString: string[];
  rolls: number[];
  description: string;
}

File: PocketEdition/RoleRollsPocketEdition/Rolls/RollInput.cs
using RoleRollsPocketEdition.Core.Entities;
using RoleRollsPocketEdition.Rolls.Entities;

namespace RoleRollsPocketEdition.Rolls
{
    public class RollInput
    {
        public int Complexity { get; set; }
        public int Difficulty { get; set; }
        public int Advantage { get; set; }
        public int Bonus { get; set; }
        public bool Hidden { get; set; }

        public string Description { get; set; }
        public List<int> Rolls { get; set; } = new List<int>();
        public Property? Property { get; set; }
        
        public Property? Attribute { get; set; }
        public int Luck { get; set; }
    }
}